src/Pact/Parse.hs

```diff
@@ -16,35 +17,31 @@
 --
 
 module Pact.Parse
-    (
-     exprs, exprsOnly
-    ,parseExprs
-    ,number
-    ,PactParser(unPactParser)
-    )
-
-where
-
-import Text.Trifecta as TF
-import Text.Trifecta.Delta as TF
-import Control.Applicative
-import Data.List
-import Control.Monad
-import Prelude
-import Data.Decimal
+  ( exprs, exprsOnly
+  , parseExprs
+  , number
+  , PactParser(unPactParser)
+  ) where
+
+import           Control.Applicative
+import           Control.Monad
 import qualified Data.Attoparsec.Text as AP
-import Data.Char (digitToInt)
-import Data.Text (Text)
-
-import Pact.Types.Exp
-import Pact.Types.Parser
-import Pact.Types.Info
+import           Data.Char (digitToInt)
+import           Data.Decimal
+import           Data.List
+import           Data.Text (Text)
+import           Prelude
+import           Text.Trifecta as TF
+import           Text.Trifecta.Delta as TF
 
-type P a = forall m. (Monad m,TokenParsing m,CharParsing m,DeltaParsing m) => PactParser m a
+import           Pact.Types.Exp
+import           Pact.Types.Parser
+import           Pact.Types.Info
 
+---
 
 -- | Main parser for Pact expressions.
-expr :: P (Exp Parsed)
```

Comment:
Why on earth are you getting rid of the type synonym????? Revert every change like this.

Reply:
There was good reason - it was only used in a few places in this module, and not of all the constraints were necessary for each function that `P` was applied to.